stackrox/stackrox
# Render policy lifecycle stage as text in Configuration Management

## Description

### Problem

Although the labels have sufficient color contrast, color text is an accessibility problem for some people, especially if meaning of hue is arbitrary (in contrast to temperature scale from cool to hot for severity).

For example, primary blue for **Build** and tertiary blue for **Runtime** is a subtle distinction. It looks like they were designed for user interface elements in different contexts instead of for a semantic contrast in the same context.

Someone has even said that primary blue for **Build** and secondary purple for **Deploy** was hard to distinguish at the size of a screen shot in a Slack channel.

Furthermore, the occurrences of secondary are rare enough to eliminate its use.

### Analysis

PatternFly policies use plain text for lifecycle stages.

### Solution

Replace `LifecycleStageLabel` element with `formatLifecycleStages` function.

Although I prefer not to import across containers, I prefer even less to refactor for the sake of a container that has been abandoned.

## Checklist

## Testing Performed

### Integration testing

* cypress/integration/configmanagement/dashboard.test.js
* cypress/integration/configmanagement/deployments.test.js
* cypress/integration/configmanagement/policies.test.js

### Manual testing

1. Visit /main/configmanagement/policies

    * `LifecycleStageLabel` elements
        ![Policies_element](https://user-images.githubusercontent.com/11862657/231288849-cc693326-1d17-4560-90f0-a5e122fba266.png)

    * `formatLifecycleStages` function
        ![Policies_function](https://user-images.githubusercontent.com/11862657/231289003-fcb60906-d822-4597-92a9-bc29f0281bee.png)

2. Click a policy

    * `LifecycleStageLabel` elements
        ![Policy_element](https://user-images.githubusercontent.com/11862657/231288777-56da519e-ee20-44c9-a6db-f4b3c6474f5a.png)

    * `formatLifecycleStages` function
        ![Policy_function](https://user-images.githubusercontent.com/11862657/231288747-89124913-6ab5-45db-9134-7e7b9c39c373.png)

3. Visit /main/violations and notice a deployment for which a policy failed

4. Visit /main/configmanagement/deployments and select that deployment

    * `LifecycleStageLabel` elements
        ![FailedPoliciesAcrossDeployment_element](https://user-images.githubusercontent.com/11862657/231288726-f2402d8b-99bb-4495-b935-84183d94b07f.png)

    * `formatLifecycleStages` function
        ![FailedPoliciesAcrossDeployment_function](https://user-images.githubusercontent.com/11862657/231288704-dc88ba3a-ea52-47eb-91c7-3e8fce3c3fc1.png)
